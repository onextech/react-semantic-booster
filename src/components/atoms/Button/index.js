import React from 'react';
import styled from 'styled-components';
import { Button as SuiButton } from 'semantic-ui-react';
import { setCustomProps } from '../../../utils/helpers';

const TEXT_CLASS = 'text';

const StyledSuiButton = styled(SuiButton)`
  &.ui.button {
    &.${TEXT_CLASS} {
      font-weight: normal;
      background-color: transparent;
      padding: 0;
    }
  }
`;

const Button = (rawProps) => {
  const props = setCustomProps(rawProps, {
    text: [TEXT_CLASS, false],
  });
  return (
    <StyledSuiButton {...props} />
  );
};

export default Button;
