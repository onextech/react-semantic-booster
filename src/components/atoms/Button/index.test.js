import React from 'react';
import renderer from 'react-test-renderer';
import Button from '.';


describe('<Button>', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<Button text content='Hello' />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
