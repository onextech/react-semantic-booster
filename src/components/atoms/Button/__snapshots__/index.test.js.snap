// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button> renders correctly 1`] = `
<button
  className="ui button text Button__StyledSuiButton-btyXTW cNVIpi"
  disabled={undefined}
  onClick={[Function]}
  role="button"
  tabIndex={undefined}
>
  Hello
</button>
`;
